### PR TITLE
build(devtools): Extract test/example app into a separate package

### PR DESCRIPTION
**Pull request by Josmithr**
Extracts the test app out as a stand-alone example app under `devtools`. We can use this as a formal example for documentation purposes in addition to testing.

### CODE REVIEW

File: packages/tools/devtools/devtools-example/.npmignore
Comment (alexvy86) [inline packages/tools/devtools/devtools-example/.npmignore:1]: Oh, I meant to ask how did we come up with this `.npmignore`? Not sure if it's that relevant in an example package, but for consistency with all the rest I think it should have an entry for `*.log` and use `**/*.tsbuildinfo`.

File: packages/tools/devtools/devtools-view/api-report/devtools-view.api.md [hunk 1]
```diff
@@ -17,4 +20,11 @@ export interface DevtoolsPanelProps {
 
 export { IMessageRelay }
 
+// @public
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/api-report/devtools-view.api.md:23]: Should we make this `@internal`? I don't think we really intend for others to implement their own visualization at this stage.

File: packages/tools/devtools/devtools-example/.npmignore
Comment (Josmithr) [reply packages/tools/devtools/devtools-example/.npmignore:1]: At this point, I don't remember 😅 I copied it from another package and removed entries that didn't appear to be relevant to the package. What generates files that match `*.log`?

File: packages/tools/devtools/devtools-view/api-report/devtools-view.api.md (hunk 1 above)
Comment (Josmithr) [reply packages/tools/devtools/devtools-view/api-report/devtools-view.api.md:23]: Yes, we should. Good call. I'll update this.

File: packages/tools/devtools/devtools-example/.npmignore
Comment (alexvy86) [reply packages/tools/devtools/devtools-example/.npmignore:1]: Not sure where they come from, but see:

image

Comment (Josmithr) [reply packages/tools/devtools/devtools-example/.npmignore:1]: Hmm. Okay. I'll add them to be safe.

File: packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts [hunk 2]
```diff
@@ -25,6 +25,8 @@ import {
  * to communicate with the debugger registry.
  * This ensures that we don't "abuse" the power of local interaction to do things that might not be possible (or need
  * to be done differently) with a message passing mechanism that crosses the boundary of the window.
+ *
+ * @internal
```
Comment (WayneFerrao) [inline packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts:29]: Why is this needed?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -42561,7 +42682,7 @@ packages:
     dependencies:
       react: 17.0.2
       react-dom: 17.0.2(react@17.0.2)
-    dev: true
+    dev: false
```
Comment (WayneFerrao) [inline pnpm-lock.yaml:42685]: How come this is being set to false?

File: packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts (hunk 2 above)
Comment (alexvy86) [reply packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts:29]: It has no functional purpose, but it's something we can do to indicate to consumers of the package "this class is not exposed for you to use directly or subclass it; it's (kind-of) an implementation detail that we need to export to make things work but it's not part of the public API of the package that you can trust".

File: pnpm-lock.yaml (hunk 3 above)
Comment (alexvy86) [reply pnpm-lock.yaml:42685]: It's probably because when `re-resizable` was a dependency of `devtools-view`, it was a dev-dependency. Now in `devtools-example` it's a regular dependency, because `devtools-example` as a whole is intended to be just an example, not a package to be published and consumed by others like `devtools-view`, so it's not an issue that it uses regular dependencies to be able to showcase what it wants to.

**Review by alexvy86 — APPROVED** [review]

File: packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts (hunk 2 above)
Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts:29]: In the future, we will leverage API-Extractor to omit such items from the typings rollup we expose publicly with the package.

File: pnpm-lock.yaml (hunk 3 above)
Comment (Josmithr) [reply pnpm-lock.yaml:42685]: Yup!

File: packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts (hunk 2 above)
Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/WindowMessageRelay.ts:29]: As for why the change was made: we didn't previously export this particular IMessageRelay implementation, but it is a generally useful one, and is used by the example app.
